crates/ruff_linter/src/rules/flake8_pyi/rules/complex_assignment_in_stub.rs

```diff
@@ -20,18 +20,28 @@ use crate::checkers::ast::Checker;
 ///
 /// ## Example
 /// ```python
+/// from typing import TypeAlias
+///
 /// a = b = int
-/// a.b = int
+///
+///
+/// class Klass:
+///     ...
+///
+///
+/// Klass.X: TypeAlias = int
 /// ```
 ///
 /// Use instead:
 /// ```python
+/// from typing import TypeAlias
+///
 /// a: TypeAlias = int
 /// b: TypeAlias = int
 ///
 ///
-/// class a:
-///     b: int
+/// class Klass:
+///     X: TypeAlias = int
```

Comment:
Noob question. Is this preferred over `type X = int` or is that syntax not allowed in classes/stubs?

Reply:
`type X = int` is allowed in classes/stubs, but it's new syntax that's only available on Python 3.12+ -- I think it's better to use syntax here that all users of Python will be able to use